cpu.cpp

```diff
@@ -354,15 +388,80 @@ void Cpu::operandFetch()
 
 void Cpu::execute()
 {
-    halt = true;
-    qDebug() << "EX I1";
-    return;
+    switch (instructionClass) {
+    case InstructionClass::b1: {
+        switch (IR >> 12) {
+        case 0:
+            mov();
+            break;
+        case 1:
+            add();
+        }
+    }
+    default:
+        break;
+    }
 }
 
 void Cpu::interrupt()
 {
    qDebug() << "INT I1";
-    return;
+   return;
+}
+
+void Cpu::mov()
+{
+    switch (cgb->getAndIncrementImpulse()) {
+    case 1: {
+        SBUS = T;
+        emit PdTS(true);
+        emit ALU(true, true, false, "SBUS");
+
+        RBUS = SBUS;
+        emit PdALU(true);
+
+        switch (mad) {
+        case AD: {
+            u8 index = IR & 0xF;
+            R[index] = RBUS;
+            emit PmRG(true, index, R[index]);
+
+            decideNextPhase();
+            break;
+        }
+        case AI: case AX:
+            MDR = RBUS;
+            emit PmMDR(true, MDR, true);
+            break;
+        }
+
+        qDebug() << "EX MOV I1";
+        break;
+    }
+    case 2: {
+        memory[ADR] = MDR & 0xFF;
+        memory[ADR + 1] = (u8)MDR >> 8;
```

Comment:
You should cast the whole shift as you will lose the high order byte of MDR by casting it to u8

Reply:
I'd like to remove cast, as it doesn't generate any warning